modules/cpr/src/main/java/org/atmosphere/interceptor/AtmosphereResourceLifecycleInterceptor.java

```diff
@@ -141,24 +141,26 @@ public void postInspect(final AtmosphereResource r) {
             && impl.isInScope()) {
 
             logger.trace("Marking AtmosphereResource {} for suspend operation", r.uuid());
-            r.addEventListener(new OnBroadcast() {
-                @Override
-                public void onBroadcast(AtmosphereResourceEvent event) {
-                    switch (r.transport()) {
-                        case JSONP:
-                        case AJAX:
-                        case LONG_POLLING:
-                            break;
-                        default:
+
+            switch (r.transport()) {
+                case JSONP:
+                case AJAX:
+                case LONG_POLLING:
+                    break;
+                default:
+                    r.addEventListener(new OnBroadcast() {
+                        @Override
+                        public void onBroadcast(AtmosphereResourceEvent event) {
                             try {
                                 r.getResponse().flushBuffer();
                             } catch (IOException e) {
                                 logger.trace("", e);
                             }
-                            break;
-                    }
-                }
-            }).suspend(timeoutInMilli);
+                        }
+                    });
+            }
+
+            r.suspend(timeoutInMilli);
```

Comment:
AFAICT, this is functionally the same thing, it just avoids adding an `OnBroadcast` event listener for `JSONP`, `AJAX` and `LONG_POLLING`.